src/Makefile

```diff
@@ -334,6 +334,8 @@ ifneq ($(SKIP_VEC_SETS),yes)
 	vpath %.c ../modules/vector-sets
 	REDIS_VEC_SETS_OBJ=hnsw.o vset.o vset_config.o
 	FINAL_CFLAGS+=-DINCLUDE_VEC_SETS=1
+	# Special compilation flags for vector-sets with AVX optimizations
+	VEC_SETS_CFLAGS=$(FINAL_CFLAGS) -mavx512f -mavx2 -mfma -march=native
```

Comment:
i would like to determine if using it at the compile time instead forcing it.
we might compile with the CPU that doesn't support this optimizetion.
refer to https://github.com/slice4e/redis/issues/13558

Reply:
Thank you for reviewing. Will look into that. The link that you shared is not working for me? Can you please confirm

Reply:
@slice4e sorry, update the link.

Reply:
Thank you @sundb . I implemented proposed changes. Agree - I had hardcoded the building just for testing purposes. Can you please take a look now and let me know what you think? thank you